File: nodejs/faturamento.js
const faturamento = require("../dados.json");

const valores = faturamento
  .filter((dia) => dia.valor > 0)
  .map((dia) => dia.valor);

// Calculando o menor, maior valor e a média
const menor = Math.min(...valores);
const maior = Math.max(...valores);
const media = valores.reduce((acc, val) => acc + val, 0) / valores.length;

// Contando os dias com faturamento superior ? média
const diasAcimaMedia = valores.filter((valor) => valor > media).length;

console.log(`Menor valor de faturamento: R$${menor.toFixed(2)}`);
console.log(`Maior valor de faturamento: R$${maior.toFixed(2)}`);
console.log(`Dias com faturamento superior a media: ${diasAcimaMedia}`);

File: python/faturamento.py
import json

with open("dados.json", "r") as file:
    faturamento = json.load(file)

valores = [dia["valor"] for dia in faturamento if dia["valor"] > 0]
menor = min(valores)
maior = max(valores)
media = sum(valores) / len(valores)

dias_acima_media = len([v for v in valores if v > media])

print(f"Menor valor: {menor}")
print(f"Maior valor: {maior}")
print(f"Dias acima da média: {dias_acima_media}")
